### PR TITLE
[W5.11][T12-1] James Arista Yaputra

### PR DESCRIPTION
**Pull request by jamesyaputra**

### CODE REVIEW

**Comment by Geraldcdx** [comment]
Good job James! I think you spend a lot of time writing the code. The incremental commits allowed me to track the changes easily! Some comments would help for me to better understand the code you have written!

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -91,6 +91,27 @@ Returns `John Doe` but not `john`.
...
+
+Put a `p` before the phone / email / address prefixes to mark it as `private`. `private` details can only
+be seen using the `viewall` command.
+
+Persons can have any number of tags (including 0).
+****
+
+Examples:
+
+* `list` +
```
Comment (devamanyu) [inline docs/UserGuide.adoc:111]: Good job in making an elaborate update to the user guide!

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -78,6 +87,9 @@ public Command parseCommand(String userInput) {
         case AddCommand.COMMAND_WORD:
             return prepareAdd(arguments);
 
+            case EditCommand.COMMAND_WORD:
```
Comment (devamanyu) [inline src/seedu/addressbook/parser/Parser.java]: Minor coding style violation.

File: src/seedu/addressbook/commands/EditCommand.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.person.*;
```
Comment (devamanyu) [inline src/seedu/addressbook/commands/EditCommand.java:6]: Minor coding style violation. Avoid performing batch import.

File: src/seedu/addressbook/commands/EditCommand.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, toEdit));
+        } catch (UniquePersonList.DuplicatePersonException dpe) {
+            return new CommandResult(MESSAGE_DUPLICATE_PERSON);
+        } catch (IndexOutOfBoundsException ie) {
+            return new CommandResult(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        } catch (PersonNotFoundException pnfe) {
+            return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+        }
+    }
+}
```
Comment (devamanyu) [inline src/seedu/addressbook/commands/EditCommand.java:75]: Overall, this class is a good implementation! Good job.

File: test/input.txt [hunk 5]
```diff
@@ -140,6 +140,38 @@
...
+  edit 2 n/Betsy Choo pp/333333 e/benchoochoo@nus.edu.sg a/333, omega street t/secretive t/foes
+  list
+  edit 2 n/Betsy Choo p/333333 pe/benchoochoo@nus.edu.sg a/333, omega street t/secretive t/foes
+  list
+  edit 2 n/Betsy Choo p/333333 e/benchoochoo@nus.edu.sg pa/333, omega street t/secretive t/foes
+  list
+
+  # should not allow adding duplicate persons
+  edit 1 n/Betsy Choo p/333333 e/benchoochoo@nus.edu.sg pa/333, omega street t/secretive t/foes
+
```
Comment (devamanyu) [inline test/input.txt:174]: Good job for handling the test cases!

**Review by devamanyu — APPROVED** [review]
This is a good implementation. Your commits are also incremental, so Kudos for doing that!

Some minor coding style violations are present. Close this PR after reading the comments.

**Pull request closed without merging**